taskfiles/utils-cpp-lint.yaml

```diff
@@ -6,34 +6,284 @@ shopt: ["globstar"]
 tasks:
   # Runs clang-format on C++ files at the given paths.
   #
-  # @param {string} FLAGS Any flags to pass to clang-format.
-  # @param {string[]} SRC_PATHS The paths on which to run clang-format.
-  # @param {string} VENV_DIR Python virtual environment where clang-format is installed.
+  # @param {string[]} ROOT_PATHS The paths on which to start `find`. When combined with
+  # EXCLUDE_PATTERNS and INCLUDE_PATTERNS `find` will output a list of files used as the input to
+  # `clang-format`.
+  # @param {string} VENV_DIR Python virtual environment where `clang-format` is installed.
+  # @param {string[]} [EXCLUDE_PATTERNS] `-path` wildcard patterns to prune in `find`.
+  # @param {string[]} [EXTENSION_PATTERNS=["*.cpp", "*.h", "*.hpp"]] `-name` wildcard patterns to
+  # match in `find`.
+  # @param {string} [FLAGS] Any flags to pass to `clang-format`.
+  # @param {string[]} [INCLUDE_PATTERNS] `-path` wildcard patterns to match in `find`.
   clang-format:
     internal: true
+    vars:
+      EXCLUDE_PATTERNS:
+        ref: "default (list) .EXCLUDE_PATTERNS"
+      EXTENSION_PATTERNS:
+        ref: >-
+          default (list "*.cpp" "*.h" "*.hpp") .EXTENSION_PATTERNS
+      FLAGS:
+        ref: "default (list) .FLAGS"
+      INCLUDE_PATTERNS:
+        ref: "default (list) .INCLUDE_PATTERNS"
     requires:
-      vars: ["FLAGS", "SRC_PATHS", "VENV_DIR"]
+      vars: ["ROOT_PATHS", "VENV_DIR"]
     cmd: |-
```

Comment:
To be more readable, can we use `>-` and then get rid of the escaped line breaks?

Something like:
```yaml
. "{{.VENV_DIR}}/bin/activate
&& find
   {{- range .ROOT_PATHS}} ...
```

Reply:
Unfortunately, indentation causes newlines to not fold in yaml, so there is no way to indent the text without causing shell syntax errors.